File: typical90/C++/076.cpp
#include <bits/stdc++.h>
using namespace std;
#define rep(i,n) for (int i = 0; i < (n); ++i)
using ll = long long;

int main() {
    ll n; cin >> n;
    vector<ll> a(n), b = {0};
    rep(i,n) cin >> a[i];

    rep(i,n*2) b.push_back(b.back()+a[i%n]);

    const ll whole = b[n];

    auto solve = [&]() -> bool {
        if (whole % 10 != 0) return false;
        rep(i,n) {
            const ll target = b[i] + whole/10;
            if (binary_search(b.begin(), b.end(), target)) return true;
        }
        return false;
    };
    cout << (solve() ? "Yes" : "No") << endl;
    return 0;
}

File: typical90/Rust/076.rs
use proconio::input;

fn main() {
    input!{
        n: usize,
        a: [i64; n],
    }

    let mut b = vec![0; n*2+1];
    for i in 0..n*2 {
        b[i+1] += b[i] + a[i%n];
    }

    let mut whole = b[n];
    let solve = || -> bool {
        if whole % 10 != 0 {
            return false;
        }
        for i in 0..n {
            let t = b[i] + whole / 10;
            if b.binary_search(&t).is_ok() {
                return true;
            }          
        }
        false
    };

    if solve() {
        println!("Yes");
    } else {
        println!("No");
    }
}